swupd/create_manifests.go

```diff
@@ -52,18 +52,23 @@ func getOldManifest(path string) (*Manifest, error) {
 	return ParseManifestFile(path)
 }
 
-func initBundles(ui UpdateInfo, c config) ([]*Manifest, error) {
-	var err error
+func initBundles(ui UpdateInfo, c config, numWorkers int) ([]*Manifest, error) {
 	var wg sync.WaitGroup
-	workers := len(ui.bundles)
-	wg.Add(workers)
+	if numWorkers > len(ui.bundles) {
```

Comment:
what if `numWorkers == 0`?

Reply:
it is never 0 since that condition is already handled at program start. If it is 0 mixer will crash in multiple points, it won't even make it this far.

Reply:
So this is covered by setWorkers(), which checks if builder.<type>NumWorkers is < 1, and then sets it to numCPU() if it is. I'm not sure if we have to check that again in here, since this function is likely not going to get called by any other thing individually.
I would have like numWorkers to be able to be > len(ui.Bundles), but since we actually do IO here we probably can't do that.

Reply:
OK, but see my additional review comment on `CreateManifests`. This is not the only way to get this point.

Reply:
E.g.: resolveFiles() and resolvePackages() inside buildBundles() also rely on workers > 0, but they are executed way earlier in the build process.

Reply:
that's fine, those are all internal functions not exposed as part of the API.

Reply:
Not directly, but they are exposed by the builder package and there is no check in the execution path.

Reply:
I have issues with the builder package. I think it should be an internal package instead of exposed at all, since it is only intended to be used via the mixer cmdline. 🤷‍♂️ But yes, it definitely violates the principles of a good external API (but probably shouldn't be one).